lib/formatter/csv/annotation_for_csv.rb

```diff
@@ -6,34 +6,100 @@ class AnnotationForCsv
         @classification = classification
         @annotation = annotation.dup.with_indifferent_access
+        @current = @annotation.dup
         @cache = cache
       end
 
       def to_h
-        annotation.merge! "task_label" => task_label
-
-        case task["type"]
+        case task['type']
         when "drawing"
-          value_with_tool = (annotation["value"] || []).map do |drawn_item|
-            drawn_item.merge "tool_label" => tool_label(drawn_item)
-          end
-          annotation.merge!("value" => value_with_tool)
-        when /single/i
-          annotation.merge!("value" => answer_label)
-        when /multiple/i
-          annotation.merge!("value" => answer_labels)
+          drawing
+        when "single", "multiple"
+          simple
+        when "text"
+          text
+        when "combo"
+          combo
+        when "dropdown"
+          dropdown
+        else
+         { error: "task cannot be exported" }
         end
-        annotation
       end
 
       private
 
-      def task_label
-        translate(task["question"] || task["instruction"])
+      def drawing(task_info=task)
+        {}.tap do |new_anno|
+          new_anno['task'] = @current['task']
+          new_anno['task_label'] = task_label(task_info)
+          value_with_tool = (@current["value"] || []).map do |drawn_item|
+           drawn_item.merge "tool_label" => tool_label(task_info, drawn_item)
+          end
+          new_anno["value"] = value_with_tool
+        end
+      end
+
+      def simple(task_info=task)
+        {}.tap do |new_anno|
+          new_anno['task'] = @current['task']
+          new_anno['task_label'] = task_label(task_info)
+          new_anno['value'] = task['type'] == 'multiple' ? answer_labels : answer_label
+        end
+      end
+
+      alias_method :single, :simple
+      alias_method :multiple, :simple
+
+      def text(task_info=task)
+        {}.tap do |new_anno|
+          new_anno['task'] = @current['task']
+          new_anno['value'] = @current['value']
+          new_anno['task_label'] = task_label(task_info)
+        end
+      end
+
+      def combo
+        {}.tap do |new_anno|
+          new_anno['task'] = annotation['task']
+          new_anno['task_label'] = nil
+          new_anno['value'] ||= []
+          annotation['value'].each do |subtask|
+            @current = subtask
+            task_info = get_task_info(subtask)
+            new_anno['value'].push send(task_info['type'], task_info)
+          end
+        end
       end
 
-      def tool_label(drawn_item)
-        tool = task["tools"] && task["tools"][drawn_item["tool"]]
+      def dropdown
+        {}.tap do |new_anno|
+          new_anno['task'] = @current['task']
+          new_anno['value'] = []
+          task['selects'].each_with_index do |sel, idx|
+            {}.tap do |tmp|
+              tmp['select_label'] = sel['title']
+              tmp['option'] = sel['allowCreate']
+              sel['options'].each do |key, arr|
```

Comment:
Definitely my least favorite part, but I was trying to just get it working. The nesting is due to the unfortunate way the data is structured. I'm sure there's a clever way to do this, I'll think on how to make it more readable. Suggestions appreciated.